app/plugin_api.go

```diff
@@ -650,3 +650,33 @@ func (api *PluginAPI) LogError(msg string, keyValuePairs ...interface{}) {
 func (api *PluginAPI) LogWarn(msg string, keyValuePairs ...interface{}) {
 	api.logger.Warn(msg, keyValuePairs...)
 }
+
+func (api *PluginAPI) CreateBot(bot *model.Bot) (*model.Bot, *model.AppError) {
+	// Bots created by a plugin should use the plugin's ID for the creator field, unless 
+	// otherwise specified by the plugin.
+	if bot.CreatorId == "" {
```

Comment:
At the moment, there is no way to change this `CreatorId` afterwards: the `BotPatch` struct below (intentionally) does not expose a field for same. I expect this to be resolved as part of https://mattermost.atlassian.net/browse/MM-13129.